File: SigmaStudio.Server/Entities/ProjectModel.cs
namespace SigmaStudio.Server.Entities
{
    public class ProjectModel
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Slug { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string ImageUrl { get; set; } = string.Empty;
        public string GithubUrl { get; set; } = string.Empty;

        public List<string> TechStack { get; set; } = new();
        public List<string> Screenshots { get; set; } = new();
        public List<ProjectSection> Sections { get; set; } = new();
    }
}

File: sigmastudio.client/src/app/features/projects/models/project.model.ts
export interface ProjectSection {
  title: string;
  content: string;
}

export interface ProjectModel {
  id: number;
  title: string;
  slug: string;
  description: string;
  imageUrl: string;
  githubUrl: string;
  techStack: string[];
  screenshots: string[];
  sections: ProjectSection[];
}
